operator/iamkeycontroller/observe.go

```diff
@@ -53,68 +55,68 @@ func (p *IAMKeyPipeline) Observe(ctx context.Context, mg resource.Managed) (mana
 		return managed.ExternalObservation{ResourceExists: true, ResourceUpToDate: false}, nil
 	}
 
+	log.V(1).Info("Observing resource line 58")
 	iamKey.Status.AtProvider.KeyName = *pctx.iamExoscaleKey.Name
-	iamKey.Status.AtProvider.ServicesSpec.SOS.Buckets = getBuckets(*pctx.iamExoscaleKey.Resources)
```

Comment:
Now as I understand it's only one bucket from the role?

Reply:
We need to update that in the status field. Also you will have to distinguish between old keys and new keys...

Reply:
fixed in newest commit